static/src/javascripts/projects/commercial/modules/prebid/slot-config.js

```diff
@@ -22,13 +22,16 @@ const filterByAdvertId = (
     return adUnits;
 };
 
-const getMostPopularDesktopSizes = memoize((isArticle: boolean) => {
+// Note: memoize only works automagically from the first argument.
+const getMostPopularSizes = memoize((isArticle: boolean, isTablet: boolean) => {
     // Only works for articles for now.
     if (isArticle && config.get('switches.extendedMostPopular')) {
-        return [[300, 600], [300, 250]];
+        return isTablet
+            ? [[300, 600], [300, 250], [728, 90]]
+            : [[300, 600], [300, 250]];
     }
     return [[300, 250]];
-});
+}, (isArticle: boolean, isTablet: boolean) => (isArticle ? 't' : 'f') + (isTablet ? 't' : 'f'));
```

Comment:
Would it be easier to understand (and wouldn't need explanatory comment) if you put the sizes inline in the `getSlots` method?  I'm not sure how much value the memoising gives, as it's not a very complicated calculation and doesn't require external resources.

Reply:
I didn't want to repeat the logic around 'isArticle' and 'has new style most popular component'. This code will be simplified when we definitely get rid of the old most popular components.

Reply:
I agree the additional complexity here is possibly not worth the cost. It might be easier to understand two different, smaller functions (one for desktop, one for tablet)?

Alternatively, could we use a single function that holds the logic to decide if the page does have an extended most pop? Then store and use the boolean like so:

```js
        {
            key: 'mostpop',
            sizes: hasExtendedMostPop
                ? [[300, 600], [300, 250], [728, 90]]
                : [[300, 250]],
        },
```

Reply:
That would be only for articles, as for now only articles can have the new extended most pop.

Reply:
So we can use something like this to workout if mostPop is extended:

```js
const isExtendedMostPop = (contentType: string): boolean =>
    contentType === 'Article' && config.get('switches.extendedMostPopular');
```

Then store the function in a const to reuse it when we create our arrays of PrebidSlots:

```js
const getSlots = (contentType: string): Array<PrebidSlot> => {
    const hasExtendedMostPop = isExtendedMostPop(contentType);
```

Reply:
> for now only articles can have the new extended most pop

We could potentially expand this, since `Gallery`, `LiveBlog`, `Crossword`, etc. are all `contentTypes` too 🙃